### ISSUE TITLE
Use pre-commit for assuring code quality 

### ISSUE THREAD

**Issue by sfc-gh-turbaszek** [snowflakedb/gosnowflake#402]
### Issue description
 Consider using pre-commit for code formatting and linting so reviewers can focus on code logic not on code quality.
Fore example here https://github.com/snowflakedb/gosnowflake/pull/379#pullrequestreview-611873631 was a lot of comment about code quality that could be automatically enforced and checked on CI.

This can be done using pre-commit framework:
https://pre-commit.com
and golangci-lin:
https://github.com/golangci/golangci-lint

Example configuration:
https://github.com/kedacore/keda/blob/main/.golangci.yml
https://github.com/kedacore/keda/blob/b9d350145c71fbd4680062aee478660d5025b886/.pre-commit-config.yaml#L39-L44

**Comment by sfc-gh-turbaszek** [comment]
WDYT @sfc-gh-jbahk ?

**Comment by sfc-gh-jbahk** [comment]
@sfc-gh-turbaszek agreed, though at the moment there are some expectations in running `go fmt` and `make lint` which the format/linting is done automatically (the lint is detected on GitHub actions before you can even merge to master).

**Comment by sfc-gh-turbaszek** [comment]
@sfc-gh-jbahk not sure about issue workflow in this project but I don't see this issue being addressed. It's a feature request that can be addressed in future and probably a good candidate for good first issue. This if I'm not mistaken is an open source project and having such issues is crucial to grow community, unless that's not we aim for.
